src/components/icon/icon.js

```diff
@@ -1,15 +1,20 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
+import Text from './components/text';
 
 import Element from '../element';
 
 const Icon = ({ size, color, className, align, text, ...props }) => {
+  if (text !== undefined) {
+    console.warn(
+      '[Deprecation] Text prop for Icon Component its deprecated, please use Icon.Text Component instead',
+    );
+  }
   return (
     <Element
       {...props}
-      className={classnames(className, {
-        icon: !text,
```

Comment:
what about keeping the changes for bug-fixing's sake? existing users will get the fix but also a warning to use `Icon.Text` instead.